clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -181,6 +215,15 @@ public synchronized void update(Cluster cluster, long now) {
     }
 
     /**
+     * If expiry is enabled, remove the unknown topic from the metadata refresh set to
+     * avoid further metadata requests for deleted topics which are not in use.
+     */
+    public synchronized void handleUnknownTopic(String topic) {
```

Comment:
@ijuma Thank you for the suggestion. Can you take a look at the latest commit and see if it is better? As you suggested, `Metadata.update` and `Metadata.handleFailedUpdate` now handle deletion of unknown topics. Since the old `Metadata.update()` method was being used all over the place including `connect` and `streams` for initializing the metadata, I retained the method with the old signature as well. I wasn't sure if `Metadata` is considered a public interface, if so, I should probably retain the old `failedUpdate` method too.